python/tvm/topi/cuda/conv2d_alter_op.py

```diff
@@ -83,14 +83,19 @@ def _alter_conv2d_layout(attrs, inputs, tinfos, out_type):
     cfg = dispatch_ctx.query(target, workload)
     if cfg.is_fallback:  # if is fallback, clear query cache and return None
         autotvm.task.clear_fallback_cache(target, workload)
-        return None
+        if (
+            "vulkan" not in target.keys
+            or "+dotprod" not in target.mattr
+            or not target.supports_integer_dot_product
+        ):
+            return None
```

Comment:
What is this change for?

Reply:
If it is fall back and if dot product is to be used, make the flow to proceed to do the NCHW => NCHW4c layout transformation.